pennylane/workflow/qnode.py

```diff
@@ -895,6 +895,15 @@ def _impl_call(self, *args, **kwargs) -> qml.typing.Result:
         return _to_qfunc_output_type(res, self._qfunc_output, tape.shots.has_partitioned_shots)
 
     def __call__(self, *args, **kwargs) -> qml.typing.Result:
+        # Check for conflicting shots configuration
+        if "shots" in kwargs:
+            if qml.set_shots in self.transform_program:

```

Comment:
Also we should combine these to one line
```suggestion
        if "shots" in kwargs and qml.set_shots in self.transform_program:
```
Note that the warning will need to be un-indented.